pkgs/applications/editors/neovim/utils.nix

```diff
@@ -54,25 +49,20 @@ let
       };
 
       # transform all plugins into an attrset
-      pluginsNormalized = map (x: if x ? plugin then { optional = false; } // x else { plugin = x; optional = false;}) plugins;
-
+      # { optional = bool; plugin = package; dest = filename; }
+      pluginsNormalized = map (x: if x ? plugin then { dest = "init.vim"; optional = false; } // x else { plugin = x; optional = false;}) plugins;
 
-      configurePatched = configure // {
-        customRC = pluginRc + customRC + (configure.customRC or "");
-      };
 
-      # A function to get the configuration string (if any) from an element of 'plugins'
-      pluginConfig = p:
-        if (p.config or "") != "" then ''
-          " ${p.plugin.pname or p.plugin.name} {{{
-          ${p.config}
-          " }}}
-        '' else "";
 
-      pluginRc = lib.concatMapStrings pluginConfig pluginsNormalized;
```

Comment:
Removing this causes issues when you specify plugin configs as single-line strings such as:

```
{
  plugin = vim-autoswap;
  config = "let g:autoswap_detect_tmux = 1";
}
```

Reply:
It also makes the resulting config much less readable...

Reply:
I removed pluginConfig because it assumed viml (customRc could be lua) + it was (my) opiniated choice and I thought  It's best to let users handle that the way they want (they should implement/call pluginConfig themselves).
The case with oneliners in the config is indeed not tested (and should be tested with several plugins) and is a regression. If you already have a fix, please submit, I can add the test.